pkg/agent/datamodel/addons.go

```diff
@@ -4,1054 +4,12 @@
-}
-
-func getAddonsIndexByName(addons []api.KubernetesAddon, name string) int {
-	for i := range addons {
-		if addons[i].Name == name {
-			return i
-		}
-	}
-	return -1
-}
-
-// assignDefaultAddonVals will assign default values to addon from defaults, for each property in addon that has a zero value
-func assignDefaultAddonVals(addon, defaults api.KubernetesAddon, isUpgrade bool) api.KubernetesAddon {
-	if addon.Enabled == nil {
-		addon.Enabled = defaults.Enabled
-	}
-	if !to.Bool(addon.Enabled) {
-		return api.KubernetesAddon{
-			Name:    addon.Name,
-			Enabled: addon.Enabled,
-		}
-	}
-	if addon.Data != "" {
-		return api.KubernetesAddon{
-			Name:    addon.Name,
-			Enabled: addon.Enabled,
-			Data:    addon.Data,
-		}
-	}
-	if addon.Mode == "" {
-		addon.Mode = defaults.Mode
-	}
-	for i := range defaults.Containers {
-		c := addon.GetAddonContainersIndexByName(defaults.Containers[i].Name)
-		if c < 0 {
-			addon.Containers = append(addon.Containers, defaults.Containers[i])
-		} else {
-			if addon.Containers[c].Image == "" || isUpgrade {
-				addon.Containers[c].Image = defaults.Containers[i].Image
-			}
-			if addon.Containers[c].CPURequests == "" {
-				addon.Containers[c].CPURequests = defaults.Containers[i].CPURequests
-			}
-			if addon.Containers[c].MemoryRequests == "" {
-				addon.Containers[c].MemoryRequests = defaults.Containers[i].MemoryRequests
-			}
-			if addon.Containers[c].CPULimits == "" {
-				addon.Containers[c].CPULimits = defaults.Containers[i].CPULimits
-			}
-			if addon.Containers[c].MemoryLimits == "" {
-				addon.Containers[c].MemoryLimits = defaults.Containers[i].MemoryLimits
-			}
-		}
-	}
-	// For pools-specific configuration, we only take the defaults if we have zero user-provided pools configuration
-	if len(addon.Pools) == 0 {
-		for i := range defaults.Pools {
-			addon.Pools = append(addon.Pools, defaults.Pools[i])
-		}
-	}
-	for key, val := range defaults.Config {
-		if addon.Config == nil {
-			addon.Config = make(map[string]string)
-		}
-		if v, ok := addon.Config[key]; !ok || v == "" {
-			addon.Config[key] = val
-		}
-	}
-	return addon
-}
-
-func synthesizeAddonsConfig(addons []api.KubernetesAddon, addon api.KubernetesAddon, isUpgrade bool) {
-	i := getAddonsIndexByName(addons, addon.Name)
-	if i >= 0 {
-		addons[i] = assignDefaultAddonVals(addons[i], addon, isUpgrade)
-	}
-}
-
-func makeDefaultClusterAutoscalerAddonPoolsConfig(cs *ContainerService) []api.AddonNodePoolsConfig {
-	var ret []api.AddonNodePoolsConfig
-	for _, pool := range cs.Properties.AgentPoolProfiles {
-		ret = append(ret, api.AddonNodePoolsConfig{
-			Name: pool.Name,
-			Config: map[string]string{
-				"min-nodes": strconv.Itoa(pool.Count),
-				"max-nodes": strconv.Itoa(pool.Count),
-			},
-		})
-	}
-	return ret
-}
-
 // GetClusterAutoscalerNodesConfig returns the cluster-autoscaler runtime configuration flag for a nodepool
 func GetClusterAutoscalerNodesConfig(addon api.KubernetesAddon, cs *ContainerService) string {
```

Comment:
"GetClusterAutoscalerNodesConfig" is kept because pkg/agent/utils.go uses it.

Reply:
It's also not needed, I just updated utils.go.

Reply:
Thanks for updating utils.go. Code removed now.